parsers/callArgsParser.go

```diff
@@ -32,6 +34,24 @@ func (parser *callArgsParser) ParseData(data string) (string, [][]byte, error) {
 	return function, arguments, nil
 }
 
+// ParseArguments parses strings of the following format:
+// argFoo@hex(argBarHex)...
+func (parser *callArgsParser) ParseArguments(data string) ([][]byte, error) {
+	tokens := strings.Split(data, atSeparator)
+	arguments := make([][]byte, 0)
+	arguments = append(arguments, []byte(tokens[0]))
```

Comment:
so the first argument is not decoded since it might be a function call. Is this intended? 
I know this works with empty "function name"

Reply:
It is not decoded as it is a difference of hex decode from empty, 0 and an element of 1 length. This function is used in asyncCallBack. Where the first input is empty or 1 length - vmcommon.ReturnCode

Reply:
yes. it is intended because first argument in asyncCallBacks is one single byte. this is called in asyncCallback vmInput creation only.